cranelift/codegen/meta/src/shared/legalize.rs

```diff
@@ -213,6 +216,30 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
     // embedded as part of arguments), so use a custom legalization for now.
     narrow.custom_legalize(iconst, "narrow_iconst");
 
+    narrow.legalize(
+        def!(istore32.I64(flags, a, ptr, offset)),
+        vec![
+            def!((al, ah) = isplit(a)),
+            def!(store.I32(flags, al, ptr, offset)),
+        ],
+    );
+
+    expand.legalize(
+        def!(a = sload32.I32(flags, ptr, offset)),
+        vec![
+            def!(b = load.I32(flags, ptr, offset)),
+            def!(a = sextend.I64(b)),
+        ],
+    );
+
+    expand.legalize(
+        def!(a = uload32.I32(flags, ptr, offset)),
+        vec![
+            def!(b = load.I32(flags, ptr, offset)),
+            def!(a = uextend.I64(b)),
+        ],
+    );
+
```

Comment:
This uses an `I32` pointer (the controlling type) to load the low 32 bits of a register and then just extend (unsigned) to 64 bits. In 32-bit x86, I would think that `uextend.I64` should just zero out another register?

Reply:
You're correct that this could be replaced with just `iconst`, I kept it symmetric but there's no reason other than that.

Reply:
I believe that is exactly how `uextend.i64` is legalized on i686.